### PR TITLE
Chairmanager merging V3

### PR DESCRIPTION
**Pull request by thelivingdiamond**

### CODE REVIEW

File: ChairManager/ChairManager.cpp [hunk 1]
```diff
@@ -350,7 +352,7 @@ void ChairManager::DrawMainWindow(bool* pbIsOpen)
         DrawAssetView();
         DrawLog();
         DrawDLLSettings();
-#ifdef _DEBUG
+#if 1
```
Comment (tmp64) [inline ChairManager/ChairManager.cpp:355]: Should be disabled for release. Maybe hide behind a cmd line arg?

File: ChairManager/ChairManager.cpp [hunk 2]
```diff
@@ -1919,12 +1901,11 @@ bool ChairManager::packChairloaderPatch() {
         }
     }
 
-    if (doLevelPatches)
+    if (doLevelPatches) //FIXME: DEBUG STEP
```
Comment (tmp64) [inline ChairManager/ChairManager.cpp:1904]: FIXME

File: ChairManager/ChairManager.cpp [hunk 3]
```diff
@@ -1659,37 +1663,18 @@ void ChairManager::EnableMod(std::string modName, bool enabled) {
...
-//    m_State = State::MainWindow;
-    return false;
-}
-
 void ChairManager::mergeXMLFiles(bool onlyChairPatch) {
     // Remove old output files
     m_DeployLogMutex.lock();
-    m_DeployState = DeployState::RemovingOldOutput;
-    m_DeployLogMutex.unlock();
+        m_DeployLogMutex.unlock();
```
Comment (tmp64) [inline ChairManager/ChairManager.cpp:1669]: Mutex is locked and immediately unlocked? Same in multiple places nearby

File: ChairManager/ConfigManager.cpp [hunk 4]
```diff
@@ -178,10 +177,198 @@ void ConfigManager::drawXMLConfigNode(pugi::xml_node node) {
 }
 
 void ConfigManager::setDirty(const std::string& modName, bool dirty) {
+//    std::scoped_lock lock(m_modConfigsMutex);
```
Comment (tmp64) [inline ChairManager/ConfigManager.cpp:180]: Why is the mutex commented out?

File: ChairManager/Data/HashesOutput.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<files>
+    <file path="LevelOutput\Campaign\EndGame\level.pak"
+          hash="532439abc6f2dfd937c4ee2ec4f014f0da6d21d26c74af853c9b47ec63f91a3a" type="level"/>
+    <file path="LevelOutput\Campaign\Engineering\CargoBay\level.pak"
```
Comment (tmp64) [inline ChairManager/Data/HashesOutput.xml:4]: Is this a generated file? Hashes will be different on different versions of the game.

File: ChairManager/Data/ID_MatchingFiles.txt [hunk 6]
```diff
@@ -0,0 +1,85 @@
+/root/Ark/AI/AiSignalReceiverConfigs
+/root/Ark/Attention/AttentionModels.xml
```
Comment (tmp64) [inline ChairManager/Data/ID_MatchingFiles.txt:2]: Is this file used?

File: ChairManager/Data/Oldpatch/danielle_hud/power_frame_new.dds
Comment (tmp64) [inline ChairManager/Data/Oldpatch/danielle_hud/power_frame_new.dds:1]: What's "OldPatch"?

File: ChairManager/Data/Testing/Hashing/Ark.pak
Comment (tmp64) [inline ChairManager/Data/Testing/Hashing/Ark.pak:1]: Is this generated by the tests?

File: ChairManager/Data/patchChecksums.xml
Comment (tmp64) [inline ChairManager/Data/patchChecksums.xml:1]: Generated in runtime?

File: ChairManager/main.cpp [hunk 7]
```diff
@@ -5,11 +5,20 @@
 #include <windows.h>
 #include <cstdio>
 #include "UI.h"
+#include <gtest/gtest.h>
 
 int APIENTRY _tWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPTSTR lpCmdLine, int nCmdShow) {
```
Comment (tmp64) [inline ChairManager/main.cpp:10]: `return 0` is missing somewhere (see compiler warning)

File: vcpkg.json [hunk 8]
```diff
@@ -10,11 +10,14 @@
...
+    "luabridge",
+    "lua",
     "curlpp",
     "detours",
     "fmt",
     "libzip",
     "pugixml",
-    "gtest"
+    "gtest",
+    "libzippp"
```
Comment (tmp64) [inline vcpkg.json:21]: List is not sorted

**Review by tmp64 — CHANGES_REQUESTED** [review]

File: ChairManager/BinaryVersionCheck.h [hunk 9]
```diff
@@ -1,6 +1,10 @@
 //
 // Created by theli on 9/2/2022.
 //
+
+#ifndef CHAIRLOADER_BINARYVERSIONCHECK_H
+#define CHAIRLOADER_BINARYVERSIONCHECK_H
+
 #pragma once
```
[the remark this answers is not in the document]
Comment (thelivingdiamond) [reply ChairManager/BinaryVersionCheck.h:8]: Changed to just be the include guard

File: ChairManager/ChairManager.cpp (hunk 1 above)
Comment (thelivingdiamond) [reply ChairManager/ChairManager.cpp:355]: Fixed, but I don't think the menu really has much use outside of development. It's not particularly well maintained lol. Could clean it up in future

File: ChairManager/ChairManager.cpp (hunk 3 above)
Comment (thelivingdiamond) [reply ChairManager/ChairManager.cpp:1669]: fixed, additionally removed alot of the old deprecated code

File: ChairManager/ConfigManager.cpp (hunk 4 above)
Comment (thelivingdiamond) [reply ChairManager/ConfigManager.cpp:180]: Commented out bc of deadlock when setDirty is called in the draw function. Not ideal but it's part of a recursive call chain which is hard to mutex.

File: ChairManager/Data/HashesOutput.xml (hunk 5 above)
Comment (thelivingdiamond) [reply ChairManager/Data/HashesOutput.xml:4]: these are the hashes that are for a natural state for the chairloader provided PreyFiles. After 1 merge they will fall back in line with the provided files. May need to be rethought when/if we switch to preditor-unpack

File: ChairManager/Data/ID_MatchingFiles.txt (hunk 6 above)
Comment (thelivingdiamond) [reply ChairManager/Data/ID_MatchingFiles.txt:2]: it's automatically generated by the jupyter notebook

File: ChairManager/Data/Testing/Hashing/Ark.pak
Comment (thelivingdiamond) [reply ChairManager/Data/Testing/Hashing/Ark.pak:1]: Yes

File: ChairManager/Data/Oldpatch/danielle_hud/power_frame_new.dds
Comment (thelivingdiamond) [reply ChairManager/Data/Oldpatch/danielle_hud/power_frame_new.dds:1]: the texture assets with a chair hidden that were oh so divisive.

File: ChairManager/Data/patchChecksums.xml
Comment (thelivingdiamond) [reply ChairManager/Data/patchChecksums.xml:1]: Nope, see above but these are the expected chairloader output for the given PreyFiles.

File: ChairManager/main.cpp (hunk 7 above)
Comment (thelivingdiamond) [reply ChairManager/main.cpp:10]: fixed

File: vcpkg.json (hunk 8 above)
Comment (thelivingdiamond) [reply vcpkg.json:21]: fixed

File: ChairManager/Data/Oldpatch/danielle_hud/power_frame_new.dds
Comment (thelivingdiamond) [reply ChairManager/Data/Oldpatch/danielle_hud/power_frame_new.dds:1]: May delete, we should discuss what assets should be included with the chairloader patch though

File: ChairManager/Data/Testing/Hashing/Ark.pak
Comment (tmp64) [reply ChairManager/Data/Testing/Hashing/Ark.pak:1]: It should be in .gitignore then

File: ChairManager/Data/HashesOutput.xml (hunk 5 above)
Comment (tmp64) [reply ChairManager/Data/HashesOutput.xml:4]: It's fine for now. The hashes can later be generated by preditor

File: ChairManager/Data/Testing/Hashing/Ark.pak
Comment (thelivingdiamond) [reply ChairManager/Data/Testing/Hashing/Ark.pak:1]: Fixed

**Review by tmp64 — APPROVED** [review]

**Pull request merged**